config-model/src/main/java/com/yahoo/vespa/model/application/validation/Validation.java

```diff
@@ -103,4 +108,20 @@ private static void validateFirstTimeDeployment(VespaModel model, DeployState de
         new AccessControlOnFirstDeploymentValidator().validate(model, deployState);
     }
 
+    private static void deferConfigChangesForClustersToBeRestarted(List<ConfigChangeAction> actions, VespaModel model) {
+        Set<ClusterSpec.Id> clustersToBeRestarted = actions.stream()
+                                                           .filter(action -> action.getType() == ConfigChangeAction.Type.RESTART)
+                                                           .map(action -> action.clusterId())
+                                                           .collect(Collectors.toSet());
+        for (var clusterToRestart : clustersToBeRestarted) {
+            var containerCluster = model.getContainerClusters().get(clusterToRestart.value());
```

Comment:
Wont this throw NPE?

Reply:
Yes, I forgot about that. Excellent catch!